backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -677,37 +690,97 @@ open class SubmissionDatabaseService(
             SequenceEntriesView.statusIsOneOf(listOfStatuses)
         }
 
-        if (warningsFilter == WarningsFilter.EXCLUDE_WARNINGS) {
+        processingResultFilter?.let {
             filteredQuery.andWhere {
-                not(SequenceEntriesView.entriesWithWarnings)
+                // Filter for has errors or not, but only if status is processed
+                not(SequenceEntriesView.statusIs(Status.PROCESSED)) or
+                    (
+                        SequenceEntriesView.statusIs(Status.PROCESSED) and
```

Comment:
```suggestion
```

This condition is redundant, right?

Reply:
I think it is not entirely redundant, I have now a function like this:

```kotlin
    fun processingResultIs(processingResult: ProcessingResult) = when (processingResult) {
        ProcessingResult.ERRORS -> SequenceEntriesView.hasErrors
        ProcessingResult.NO_ISSUES -> not(
            SequenceEntriesView.hasErrors or SequenceEntriesView.hasWarnings,
        ) and SequenceEntriesView.statusIs(Status.PROCESSED)
        ProcessingResult.WARNINGS -> not(SequenceEntriesView.hasErrors) and
            SequenceEntriesView.hasWarnings
    }
```

I think for the NO_ISSUES path it is still relevant, otherwise we would lump unprocessed and processed but no issues together.

https://github.com/loculus-project/loculus/pull/3125/commits/59f73282057b80b8a297cb4a314f94b15129c208